### ISSUE TITLE
Unable to boot VM after following instructions in CONTRIBUTING.md for local BLT development setup

### ISSUE THREAD

**Issue by ba66e77** [acquia/blt#2890]
My system information:
* Operating system type: OSX
* Operating system version: 10.13.5
* BLT version: 9.x

When I follow the directions in [CONTRIBUTING.md](https://github.com/acquia/blt/blob/9.x/.github/CONTRIBUTING.md) to set up  an environment for developing BLT locally, when running `blt vm` in the resulting environment, booting the VM fails with the following errors:
```
failed: [blted8] (item=../../../acquia/blt/scripts/drupal-vm/configure-solr.sh) => {"changed": false, "item": "../../../acquia/blt/scripts/drupal-vm/configure-solr.sh", "msg": "Could not find or access '../../../acquia/blt/scripts/drupal-vm/configure-solr.sh'\nSearched in:\n\t/vagrant/vendor/geerlingguy/drupal-vm/provisioning/files/../../../acquia/blt/scripts/drupal-vm/configure-solr.sh\n\t/vagrant/vendor/geerlingguy/drupal-vm/provisioning/../../../acquia/blt/scripts/drupal-vm/configure-solr.sh\n\t/vagrant/vendor/geerlingguy/drupal-vm/provisioning/files/../../../acquia/blt/scripts/drupal-vm/configure-solr.sh\n\t/vagrant/vendor/geerlingguy/drupal-vm/provisioning/../../../acquia/blt/scripts/drupal-vm/configure-solr.sh"}
failed: [blted8] (item=../../../acquia/blt/scripts/drupal-vm/post-provision.php) => {"changed": false, "item": "../../../acquia/blt/scripts/drupal-vm/post-provision.php", "msg": "Could not find or access '../../../acquia/blt/scripts/drupal-vm/post-provision.php'\nSearched in:\n\t/vagrant/vendor/geerlingguy/drupal-vm/provisioning/files/../../../acquia/blt/scripts/drupal-vm/post-provision.php\n\t/vagrant/vendor/geerlingguy/drupal-vm/provisioning/../../../acquia/blt/scripts/drupal-vm/post-provision.php\n\t/vagrant/vendor/geerlingguy/drupal-vm/provisioning/files/../../../acquia/blt/scripts/drupal-vm/post-provision.php\n\t/vagrant/vendor/geerlingguy/drupal-vm/provisioning/../../../acquia/blt/scripts/drupal-vm/post-provision.php"}
```

It appears there are two factors causing this:
1. No NFS mount is provisioned for the symlinked blt directory, so the VM is unable to access the blt directory at all.
2. When I edit the box/config.yml to include a mount beside the path where blted8 is mounted, vagrant gets the symlinked path of '../../../blt' and still ends up looking in the wrong place.

**Comment by ba66e77** [comment]
The work around I described above is encoded in PR #2900. I don't love it, but don't have a better solution at the moment.

**Comment by ba66e77** [comment]
At least one other user has told me they do not have the same issue, even after confirming we're using the same versions of BLT, vagrant, and virtualbox.

Can anyone replicate the problems I'm seeing?

**Comment by danepowell** [comment]
Yes, I get the same error when following the same steps. For what it's worth, I don't see where CONTRIBUTING.md implies that this should work, although it obviously would be nice if it did.

It looks like BLT is actually fairly clever and does try to mount itself in Vagrant in a way that you can traverse the symlink both on the host and in the VM. But for the purposes of the post-provision script, the mount doesn't work, because the post-provision scripts execute from the `/vagrant` directory instead of `/var/www/blted8`.

Note that `/vagrant` and `/var/www/blted8` have the exact same contents (the project root directory on the host), but it appears they are mounted separately for whatever reason. It's problematic because BLT is mounted as a sibling of `/var/www/blted8` at `/var/www/blt` but not as a sibling of `/vagrant` at `/blt`, so the symlink fails.

It seems like the solution is force DrupalVM to run the scripts from `/var/www/blted8`, either by specifying the full path of the post-provision scripts in `box/config.yml`, or by changing DrupalVM's default behavior (not sure where it comes from).

For anyone else following along at home, the new location for CONTRIBUTING.md is https://github.com/acquia/blt/blob/9.x/docs/CONTRIBUTING.md

**Comment by cballenar** [comment]
Correction, third line is also to a `.sh` file. It should be:

```
post_provision_scripts:
  - "../../../acquia/blt-drupal-vm/scripts/drupal-vm/configure-solr.sh"
  - "../../../acquia/blt-drupal-vm/scripts/drupal-vm/post-provision.sh"
  ```
  
  At least that was my case with BLT 12.8.2.
